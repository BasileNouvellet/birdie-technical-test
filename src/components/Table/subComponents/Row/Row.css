.row {
    display: grid;
    grid-template-columns: 10% 30% 30% 30%;
    grid-template-rows: 2em;
    grid-template-areas: 'index variable count average-age';
    align-items: center;
}

.title {
    grid-template-rows: 3em;
    font-weight: bold;
    border-top-left-radius: 10px;
    border-top-right-radius: 10px;
    margin: 0 0 5px 0;
    box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16), 0 3px 6px rgba(0, 0, 0, 0.23);
}

.index {
    grid-area: index;
}

.variable {
    grid-area: variable;
}

.count {
    grid-area: count;
}

.averageAge {
    grid-area: average-age;
}
