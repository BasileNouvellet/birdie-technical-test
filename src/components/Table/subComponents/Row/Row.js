// @flow

import * as React from 'react';
import classnames from "classnames";

import './Row.css';

type PropsType = {|
    index: number | string,
    variable: string,
    count: string,
    averageAge: number | string,
    isTitle?: boolean,
|};

function Row(props: PropsType): React.Element<'div'> {
    const { index, variable, count, averageAge, isTitle } = props;

    const classNames = classnames({
        'row': true,
        'title': isTitle,
    });

    const style = {
        backgroundColor: index % 2 ? '#e1e1e1' : null,
    };

    return (
        <div className={classNames} style={style}>
            <div className="index">{index}</div>
            <div className="variable">{variable}</div>
            <div className="count">{count}</div>
            <div className="averageAge">{averageAge}</div>
        </div>
    );
}

Row.defaultProps = {
    isTitle: false,
};

export default Row;
