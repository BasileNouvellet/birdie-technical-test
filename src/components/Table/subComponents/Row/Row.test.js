// @flow

import * as React from 'react';
import { shallow } from 'enzyme';

import Row from './Row';

describe('Row component', () => {
  it('should shallow render without crashing', () => {
    const row = (
      <Row
        index={0}
        variable="education"
        count={100}
        averageAge={200}
      />
    );

    shallow(row);
  });

  it('should match snapshot', () => {
    const row = (
      <Row
        index={1000}
        variable="education"
        count={1000}
        averageAge={1000}
      />
    );

    const wrapper = shallow(row);
    expect(wrapper).toMatchSnapshot();
  });
});
