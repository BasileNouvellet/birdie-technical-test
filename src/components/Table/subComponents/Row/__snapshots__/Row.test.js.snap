// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Row
    averageAge={1000}
    count={1000}
    index={1000}
    isTitle={false}
    variable="education"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="cell index"
        >
          1000
        </div>,
        <div
          className="cell variable"
        >
          education
        </div>,
        <div
          className="cell count"
        >
          1000
        </div>,
        <div
          className="cell averageAge"
        >
          1000
        </div>,
      ],
      "className": "row",
      "style": Object {
        "backgroundColor": null,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1000,
          "className": "cell index",
        },
        "ref": null,
        "rendered": 1000,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "education",
          "className": "cell variable",
        },
        "ref": null,
        "rendered": "education",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1000,
          "className": "cell count",
        },
        "ref": null,
        "rendered": 1000,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1000,
          "className": "cell averageAge",
        },
        "ref": null,
        "rendered": 1000,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cell index"
          >
            1000
          </div>,
          <div
            className="cell variable"
          >
            education
          </div>,
          <div
            className="cell count"
          >
            1000
          </div>,
          <div
            className="cell averageAge"
          >
            1000
          </div>,
        ],
        "className": "row",
        "style": Object {
          "backgroundColor": null,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1000,
            "className": "cell index",
          },
          "ref": null,
          "rendered": 1000,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "education",
            "className": "cell variable",
          },
          "ref": null,
          "rendered": "education",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1000,
            "className": "cell count",
          },
          "ref": null,
          "rendered": 1000,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1000,
            "className": "cell averageAge",
          },
          "ref": null,
          "rendered": 1000,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
